Assets/Scripts/SS3D/Systems/Tile/PlacedObjects/PlacedTileObject.cs

```diff
@@ -0,0 +1,131 @@
+using FishNet;
+using FishNet.Object;
+using SS3D.Core;
+using SS3D.Logging;
+using SS3D.Systems.Tile.Connections;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SS3D.Systems.Tile
+{
+    /// <summary>
+    /// Component that is added to every tile object that is part of the tilemap. Tiles are more restrictive and need to have an origin, fixed grid position and direction to face.
+    /// </summary>
+    public class PlacedTileObject: NetworkBehaviour
+    {
+        /// <summary>
+        /// Creates a new PlacedTileObject from a TileObjectSO at a given position and direction. Uses NetworkServer.Spawn() if a server is running.
+        /// </summary>
+        /// <param name="worldPosition"></param>
+        /// <param name="dir"></param>
+        /// <param name="tileObjectSo"></param>
+        /// <returns></returns>
+        public static PlacedTileObject Create(Vector3 worldPosition, Vector2Int origin, Direction dir, TileObjectSo tileObjectSo)
+        {
+            GameObject placedGameObject = Instantiate(tileObjectSo.prefab);
+            placedGameObject.transform.SetPositionAndRotation(worldPosition, Quaternion.Euler(0, TileHelper.GetRotationAngle(dir), 0));
+
+            PlacedTileObject placedObject = placedGameObject.GetComponent<PlacedTileObject>();
+            if (placedObject == null)
+            {
+                // Ideally an editor script adds this instead of doing it at runtime
+                placedObject = placedGameObject.AddComponent<PlacedTileObject>();
+            }
+
+            placedObject.Setup(tileObjectSo, origin, dir);
+
+            if (InstanceFinder.ServerManager != null)
+            {
+                if (placedObject.GetComponent<NetworkObject>() == null)
+                    Punpun.Yell(SystemLocator.Get<TileSystem>(), $"{placedObject.NameString} does not have a Network Component and will not be spawned");
+                else
+                    InstanceFinder.ServerManager.Spawn(placedGameObject);
+            }
+
+            return placedObject;
+        }
+
+        /// <summary>
+        /// SaveObject that contains all information required to reconstruct the object.
+        /// </summary>
+        [Serializable]
+        public class PlacedSaveObject
```

Comment:
before acceptating the PR I just want to resolve this conversation.

@henkhooft : 

> This is more of a design choice to keep the "save" object in the same class as it makes a save of. It has to be public since the tilemap or chunk needs to know it's contents during loading. Let me know if you disagree.

@stilnat :

> I understand the intent, and it's a good one. However this goes against our C# style guide, one declaration per file for class structs, enums .... Could you rename it in SavedPlacedItemObject, make it a struct (for performance reasons), put it in its own file ?


Do the same for ChunkSaveObject, rename it to SavedTileChunk and in it's own file (don't make this one a struct, its memory footprint is too large.)
If you disagree, I would understand as your choice makes sense too. In this case we'll ask John to make a choice.